rotkehlchen/chain/evm/utils.py

```diff
@@ -0,0 +1,140 @@
+import logging
+from typing import TYPE_CHECKING, Any, Callable, Final, Optional
+
+from web3.types import BlockIdentifier
+
+from rotkehlchen.assets.asset import EvmToken
+from rotkehlchen.chain.evm.contracts import EvmContract
+from rotkehlchen.constants import ZERO
+from rotkehlchen.constants.resolver import evm_address_to_identifier
+from rotkehlchen.errors.asset import UnknownAsset, WrongAssetType
+from rotkehlchen.errors.misc import BlockchainQueryError, RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import UNISWAP_PROTOCOL, VELODROME_POOL_PROTOCOL, EvmTokenKind, Price
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import LP_TOKEN_AS_POOL_CONTRACT_ABIS
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+LP_TOKEN_AS_POOL_PROTOCOL_TO_ABI_NAME: Final[dict[str, 'LP_TOKEN_AS_POOL_CONTRACT_ABIS']] = {
+    VELODROME_POOL_PROTOCOL: 'VELO_V2_LP',
+    UNISWAP_PROTOCOL: 'UNISWAP_V2_LP',
+}
+MULTICALL_DEPLOYMENT_BLOCK = 7929876
```

Comment:
This is wrong here. You are using a constant for the multicall we use in mainnet. That is a mainnet block for a particular multicall version.

The code here is supposed to be **EVM chain agnostic**.

We are lucky though, since we have this data per chain. Look further down in the review.